src/webgpu/shader/validation/parse/break_if.spec.ts

```diff
@@ -0,0 +1,133 @@
+export const description = `Validation tests for break if`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kTests = {
+  loop_break: {
+    src: 'loop { break if true; }',
+    pass: false,
+  },
+  loop_if_break: {
+    src: 'loop { if true { break if false; } }',
+    pass: false,
+  },
+  continuing_break_if: {
+    src: 'loop { continuing { break if (true); } }',
+    pass: true,
+  },
+  continuing_break_if_not_last: {
+    src: 'loop { continuing { break if (true); let a = 4;} }',
+    pass: false,
+  },
+  while_break: {
+    src: 'while true { break if true; }',
```

Comment:
Suggest adding some variants in here where you keep the passing form (_[loop]_ → `continuing` → `break-if`), but swap the `loop` with `for` and `while`. e.g:

```
while true { continuing { break if true; } }
for (;;) { continuing { break if true; } }
```

Reply:
I don't feel strongly here.